src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
-            // itself. If that is the case, the road is obviously not a sliproad.
-
-            // a sliproad has to enter a road without choice
-            const auto couldBeSliproad = [](const Intersection &intersection) {
-                if (intersection.size() != 3)
-                    return false;
-                if ((intersection[1].entry_allowed && intersection[2].entry_allowed) ||
-                    intersection[0].entry_allowed)
-                    return false;
-                return true;
-            };
+    // Potential non-sliproad road (bc), leading to the intersection (c) the Sliproad (bd) shortcuts
+    const auto &next_road = intersection[*obvious];
 
-            if (couldBeSliproad(intersection_following_index_one) &&
-                intersection_node_id != intersection_node_two)
-                return 2;
-            else if (couldBeSliproad(intersection_following_index_two) &&
-                     intersection_node_id != intersection_node_one)
-                return 1;
-            else
-                return 0;
-        }
-        else
-            return 0;
-    }();
+    // The intersection `c` has to be reachable
+    if (!next_road.entry_allowed)
+    {
+        return intersection;
+    }
 
-    if (obvious_turn_index == 0)
+    // The road leading to the intersection (bc) has to continue from our source
+    if (!roadContinues(source_edge_id, next_road.eid))
+    {
         return intersection;
+    }
 
-    const auto &next_road = intersection[obvious_turn_index];
+    // Link-check for (bc) and later on (cd) which both are getting shortcutted by Sliproad
+    const auto is_potential_link = [this, next_road](const ConnectedRoad &road) {
+        if (!road.entry_allowed)
+            return false;
 
-    const auto linkTest = [this, next_road](const ConnectedRoad &road) {
-        return !node_based_graph.GetEdgeData(road.eid).roundabout && road.entry_allowed &&
-               angularDeviation(road.angle, STRAIGHT_ANGLE) <= 2 * NARROW_TURN_ANGLE &&
-               !hasRoundaboutType(road.instruction) &&
-               angularDeviation(next_road.angle, road.angle) >
-                   std::numeric_limits<double>::epsilon();
-    };
+        // Narrow turn angle for road (bd) and guard against data issues (overlapping roads)
+        auto is_narrow = angularDeviation(road.angle, STRAIGHT_ANGLE) <= 2 * NARROW_TURN_ANGLE;
+        auto same_angle = angularDeviation(next_road.angle, road.angle) //
+                          <= std::numeric_limits<double>::epsilon();
 
-    bool hasNarrow =
-        std::find_if(intersection.begin(), intersection.end(), linkTest) != intersection.end();
-    if (!hasNarrow)
-        return intersection;
+        if (!is_narrow || same_angle)
+            return false;
+
+        // Prevent from starting in or going onto a roundabout
+        auto onto_roundabout = hasRoundaboutType(road.instruction);
```

Comment:
this check should probably be done earlier, since it is a simple check on the instruction and does not require looking for an obvious index (which costs more)